Apps/Src/ReadCarCAN.c

```diff
@@ -15,21 +16,21 @@ void Task_ReadCarCAN(void *p_arg) {
 
     while (1) {
         // Check if BPS sent us a message
-        if (CANbus_Read(buffer) == SUCCESS) {
-            long msg = *((long *)(&buffer[0]));
-            if (msg == 0) {
-                car->IsRegenBrakingAllowed = OFF;
+        if (CANbus_Read(&canId, buffer, CAN_NON_BLOCKING) == SUCCESS) {
+            // If charge_enable, set regen flag
+            if (canId == CHARGE_ENABLE) {
+                car->IsRegenBrakingAllowed = (buffer[0] == 0) ? OFF : ON;
+                Contactors_Set(ARRAY, car->IsRegenBrakingAllowed);
```

Comment:
@Cam0Cow Shouldn't we be sending something to the motor to turn off the motor side Regen Braking here? Or is this handled somewhere else at the moment

Reply:
The logic for handling this is in the UpdateVelocity task. However, looking back at it, I'm not sure if actually disables regen braking when in normal (non cruise-control) mode. I don't see any specific command for disabling regenerative braking in the wavesculptor documentation, and the code doesn't try to do anything clever to make sure regen can't happen. We should definitely figure out how we want to handle it this coming workday.